File: R/binomial-pmf-cdf.R
binomial_pmf_cdf <- function(n, p, k) {
  if (p < 0 || p > 1) {
    stop("p must be in [0, 1]")
  }
  if (k < 0 || k > n) {
    stop("k must be in [0, n]")
  }

  pmf <- dbinom(k, size = n, prob = p)
  cdf <- pbinom(k, size = n, prob = p)

  list(pmf = as.numeric(pmf), cdf = as.numeric(cdf))
}

File: Julia/binomial-pmf-cdf.jl
function binomial_pmf_cdf(n, p, k)
    if p < 0 || p > 1
        error("p must be in [0, 1]")
    end
    if k < 0 || k > n
        error("k must be in [0, n]")
    end

    pmf = binomial(n, k) * (p ^ k) * ((1 - p) ^ (n - k))
    cdf = 0.0
    for i in 0:k
        cdf += binomial(n, i) * (p ^ i) * ((1 - p) ^ (n - i))
    end

    return (pmf = float(pmf), cdf = float(cdf))
end
